crates/sui-framework/tests/rewards_distribution_tests.move

```diff
@@ -249,15 +249,15 @@ module sui::rewards_distribution_tests {
         // unslashed validators too.
         // Validator 4 should get 475 SUI of rewards without slashing. 20% is slashed so she gets
         // 380 SUI of rewards.
-        assert_validator_stake_amounts(validator_addrs(), vector[294, 508, 722, 780], scenario);
+        assert_validator_self_stake_amounts(validator_addrs(), vector[294, 508, 723, 780], scenario);
```

Comment:
I suppose this and the change below are due to rounding differences? And we are pretty sure why this change happened?

Reply:
I believe these are indeed due to rounding differences, and in particular because we no longer scale the validators commission by the validator's self stake. But I want to dive into this change a bit more and be confident in _exactly_ why they before I would call this ready :)

Reply:
Dug into this a bit more this morning and have some interesting conclusions:
1. This is _not_ because of rounding issues during rewards distribution -- if I look at the numbers getting distributed to each validator during rewards distribution before and after the changes in this PR they are the exact same same for this test. 
2. What is causing this change is the logic in the test function `calculate_rewards`:
    a. We are calculating the correct SUI amount for the deposited pool tokens (`79` in this case).
    b. However, the staked amount is `80`. So the logic in that function calculates the reward amount as `80`.

Now, I'm not sure if this is correct or not. I was simply copying the logic around `withdraw_rewards_and_burn_pool_tokens`. So I'll try to understand that a bit more this morning, but I may want to chat with you about it a bit at some point :)